modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4112,10 +4494,48 @@ def iloc_mut(partition, row_internal_indices, col_internal_indices, item):
         return self.__constructor__(new_modin_frame)
 
     def sort_rows_by_column_values(self, columns, ascending=True, **kwargs):
-        new_modin_frame = self._modin_frame.sort_by(
-            0, columns, ascending=ascending, **kwargs
-        )
-        return self.__constructor__(new_modin_frame)
+        # Our algebra sort is only implemented for Engines that support virtual partitioning.
+        if _current_engine_supports_virtual_partitions():
+            new_modin_frame = self._modin_frame.sort_by(
+                0, columns, ascending=ascending, **kwargs
+            )
+            return self.__constructor__(new_modin_frame)
+        ignore_index = kwargs.get("ignore_index", False)
+        kwargs["ignore_index"] = False
+        if not is_list_like(columns):
+            columns = [columns]
+        # Currently, sort_values will just reindex based on the sorted values.
+        # TODO create a more efficient way to sort
+        ErrorMessage.default_to_pandas("sort_values")
+        broadcast_value_dict = {
+            col: self.getitem_column_array([col]).to_pandas().squeeze(axis=1)
+            for col in columns
+        }
+        # Clear index level names because they also appear in broadcast_value_dict
+        orig_index_level_names = self.index.names
+        tmp_index = self.index.copy()
+        tmp_index.names = [None] * tmp_index.nlevels
+        # Index may contain duplicates
+        broadcast_values1 = pandas.DataFrame(broadcast_value_dict, index=tmp_index)
+        # Index without duplicates
+        broadcast_values2 = pandas.DataFrame(broadcast_value_dict)
+        broadcast_values2 = broadcast_values2.reset_index(drop=True)
+        # Index may contain duplicates
+        new_index1 = broadcast_values1.sort_values(
+            by=columns, axis=0, ascending=ascending, **kwargs
+        ).index
+        # Index without duplicates
+        new_index2 = broadcast_values2.sort_values(
+            by=columns, axis=0, ascending=ascending, **kwargs
+        ).index
+
+        result = self.reset_index(drop=True).reindex(axis=0, labels=new_index2)
+        if ignore_index:
+            result = result.reset_index(drop=True)
+        else:
+            result.index = new_index1
+            result.index.names = orig_index_level_names
+        return result
```

Comment:
```suggestion
        new_modin_frame = self._modin_frame.sort_by(
            0, columns, ascending=ascending, **kwargs
        )
        return self.__constructor__(new_modin_frame)
```

I think you didn't mean to change all this code in `sort_rows_by_column_values`. I think we can revert this block.